contents/staticAnalysis/FindBugs.md

```diff
@@ -1,17 +1,74 @@
 # FindBugs
 
-Author: [Xiao Pu](https://nus-oss.github.io/cs3281-website/students/AY1617S2/xiaoPu/xiaoPu-Resume.html)
+Author: [Xiao Pu](https://nus-oss.github.io/cs3281-website/students/AY1617S2/xiaoPu/xiaoPu-Resume.html), [Shradheya Thakre](https://github.com/tshradheya)
 
 ## Overview
 
-FindBugs is a static analysis tool to find "bugs" in **Java** programme. It looks for "bug patterns" in the code and signals possible violations. For example, the bug "null pointer dereference" has the pattern — A programme declares a non-nullable variable but assigns `null` to the variable somewhere and uses it later.
+FindBugs is a static analysis tool to find "bugs" in **Java** programme. It looks for "bug patterns" in the code and signals possible violations. Potential errors are classified in four ranks:
+1. scariest,
+1. scary,
+1. troubling,
+1. of concern.
```

Comment:
Would it be clearer if this is written as 1. 2. 3. 4. even though it is rendered similarly? I went through some other book chapters and some authors wrote it as 1. 1. 1. 1. while others wrote in numerically, so I am not sure whether there is a particular standard to this.

Reply:
I thought of following the MarkDown syntax and hence 1. 1. 1. ...

Reply:
Writing it as 1. 1. 1. has an additional benefit: should you remove/add items in the middle of the list, you need not have to renumber everything else.

Reply:
I see. Thanks for pointing out.

Reply:
By numbering the points, you can remove the commas or full-stops in each point